### DISCUSSION TITLE
MRTK objects are pink in Unity 2020

### DISCUSSION THREAD

**Discussion by rkfung88** [microsoft/MixedRealityToolkit-Unity#9563] [Q&A]
I am currently working on a class assignment with Unity 2020.2.2f1 and MRTK. Once my unity project is set up and I have installed the MRTK packages into Unity, the objects such as the piano or coffee cup are completely pink when I drag them into the scene. Would anyone know the possible cause of this issue?
![Screenshot 2021-02-17 223557](https://user-images.githubusercontent.com/61769858/112732195-89d5d480-8f06-11eb-9ae3-415b6611c24a.png)

**Comment by david-c-kline** [comment]
@rkfung88,

Do you have any errors in the Unity Console window? If you see an error about not being able to find a file, your project may be experiencing a long path issue.

Unity is limited to a 260 character maximum path length. MRTK has some long sub-paths that can necessitate using a very short project root path (ex: c:\source\myproject).

Hope this helps!

**Comment by m-the-hoff** [comment]
This can be caused by not supporting a new enough version of DirectX on your machine (or in Parallels in my case).  Some shaders are marked as requiring a minimum shader version. Do you know what versions of DirectX your machine supports?  

If this is the root cause, I've noticed that some shaders are marked as requiring newer DirectX versions than they actually require and taking the pragma out of the shader code fixes the problem.  I'd be curious to know if this is your issue.
